src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -18,19 +18,92 @@
  */
 class RedisAdapter extends AbstractAdapter
 {
+    private static $defaultConnectionOptions = array(
+        'class' => \Redis::class,
+        'persistent' => 0,
+        'timeout' => 0,
+        'read_timeout' => 0,
+        'retry_interval' => 0,
+    );
     private $redis;
 
     public function __construct(\Redis $redisConnection, $namespace = '', $defaultLifetime = 0)
     {
         if (preg_match('#[^-+_.A-Za-z0-9]#', $namespace, $match)) {
             throw new InvalidArgumentException(sprintf('RedisAdapter namespace contains "%s" but only characters in [-+_.A-Za-z0-9] are allowed.', $match[0]));
         }
+        if (!$redisConnection->isConnected()) {
```

Comment:
Are you sure that it is a good idea to require an established connection in the constructor?